### PR TITLE
Support rewriter with preprocess and postprocess

### PR DESCRIPTION
**Pull request by qinezh**

### CODE REVIEW

**Comment by vwxyzh** [comment]
consider: move MarkdownSytleConfig to plugins project?

File: MarkdigEngine/Extensions/Rewriter/MarkdownLambdaObjectValidator.cs [hunk 1]
```diff
@@ -6,11 +6,11 @@
 
 namespace MarkdigEngine
 {
-    internal class MarkdownLambdaTokenValidator<TObject> : IMarkdownObjectValidator where TObject : class, IMarkdownObject
+    internal class MarkdownLambdaObjectValidator<TObject> : IMarkdownObjectValidator where TObject : class, IMarkdownObject
     {
         private Action<TObject> _validator;
 
-        public MarkdownLambdaTokenValidator(Action<TObject> validator)
+        public MarkdownLambdaObjectValidator(Action<TObject> validator)
```
Comment (vwxyzh) [inline MarkdigEngine/Extensions/Rewriter/MarkdownLambdaObjectValidator.cs]: add pre/post?

**Comment by vwxyzh** [comment]
:shipit:

File: MarkdigEngine/Extensions/Rewriter/MarkdownLambdaObjectValidator.cs (hunk 1 above)
Comment (qinezh) [reply MarkdigEngine/Extensions/Rewriter/MarkdownLambdaObjectValidator.cs]: it would be supported in next PR

**Pull request merged**